_posts/2021-07-16-hspec_hooks.markdown

```diff
@@ -375,12 +375,7 @@ More importantly - it happens in a composable manner.
 ```haskell
 spec :: Spec
 spec = do
-    let
-        provideDatabase runTest =
-            withDatabase $ \db ->
-                runTest db
```

Comment:
@parsonsmatt `around withResource` is a pretty common pattern.  Or is that level of "pointfulness" intentional / do you think it's necessary to guide the reader?

Reply:
Or way just simplify it to:

```haskell
provideDatabase runTest =
    withDatabase runTest
```
(but I guess if we do that it's already obvious that `provideDatabase = withDatabase` :shrug:)

Reply:
Yeah, it's more that I'm trying to make it super explicit what all the function parameters are, in an evolving manner. I know that I personally found it kind of hard to figure out how to track the function arguments in `around` and especially `aroundWith`.

I know most Haskellers can intuitively eta reduce these things, but for those newer to the language, I think it's nice to show it explicitly. Especially since any addition to the code would require eta-expanding the arguments out again.

Reply:
I guess that makes sense.